src/tcgwars/logic/impl/gen8/RebelClash.groovy

```diff
@@ -936,9 +936,14 @@ public enum RebelClash implements LogicCardInfo {
         weakness M
         bwAbility "Playful", {
           text "Once during your turn, when you play this card from your hand to evolve a Pokemon, you may choose an Energy attached to your  opponent’s Active Pokemon and return it to the top of your opponent’s deck."
-          actionA {
-            // TODO
-          }
+          onActivate {r->
+              if(r==PLAY_FROM_HAND && opp.active.cards.filterByType(ENERGY)) {
```

Comment:
This if block should be shifted back by a few spaces, if you haven't already you should set your ide to use 2 space indenting